src/core/net/mdns.hpp

```diff
@@ -1189,32 +1190,19 @@ class Core : public InstanceLocator, private NonCopyable
 
     // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-    class TimerContext : public InstanceLocator
-    {
-    public:
-        TimerContext(Instance &aInstance);
-
-        TimeMilli GetNow(void) const { return mNow; }
-        TimeMilli GetNextTime(void) const { return mNextTime; }
-        void      UpdateNextTime(TimeMilli aTime);
-
-    private:
-        TimeMilli mNow;
-        TimeMilli mNextTime;
-    };
-
-    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-
-    class EntryTimerContext : public TimerContext // Used by `HandleEntryTimer`.
+    class EntryTimerContext : public InstanceLocator // Used by `HandleEntryTimer`.
     {
     public:
         EntryTimerContext(Instance &aInstance);
-        TxMessage &GetProbeMessage(void) { return mProbeMessage; }
-        TxMessage &GetResponseMessage(void) { return mResponseMessage; }
+        TimeMilli     GetNow(void) const { return mNextFireTime.GetNow(); }
```

Comment:
It seems that the `GetNow` method of `EntryTimerContext` is not used in this PR. Is it required?

Reply:
Yes, we definitely need the `GetNow()` method. It's used in many places throughout the existing code (unaffected by this PR), such as in `Core::Entry::HandleTimer(EntryTimerContext &aContext)`.  It was previously provided by the `TimerContext` base class.

Reply:
Got it. Thanks!